### ISSUE TITLE
DMListener attempts to start async task prior to event loop creation

### ISSUE THREAD

**Issue by archfan7411** [2br-2b/StoryBot#12]
Full traceback, presently testing on Windows 10 with Python 3.10:

```
C:\Users\<redacted>\Documents\StoryBot>py main.py
Current user retrieved: `<redacted>`
Traceback (most recent call last):
  File "C:\Users\<redacted>\Documents\StoryBot\main.py", line 38, in <module>
    dml = dmlistener.dmlistener(f, u, bot)
  File "C:\Users\<redacted>\Documents\StoryBot\dmlistener.py", line 23, in __init__
    self.timeout_checker.start()
  File "C:\Users\<redacted>\AppData\Local\Programs\Python\Python310\lib\site-packages\discord\ext\tasks\__init__.py", line 398, in start
    self._task = asyncio.create_task(self._loop(*args, **kwargs))
  File "C:\Users\<redacted>\AppData\Local\Programs\Python\Python310\lib\asyncio\tasks.py", line 336, in create_task
    loop = events.get_running_loop()
RuntimeError: no running event loop
sys:1: RuntimeWarning: coroutine 'Loop._loop' was never awaited
```

L38 of `main.py` creates a DMListener before the bot is run (and the async event loop started):
```
dml = dmlistener.dmlistener(f, u, bot)
```

L23 of `dmlistener.py` starts the `timeout_checker` async task, from within the sync context of `__init__`.
```
self.timeout_checker.start()
```

This leads to the error seen within my traceback, `RuntimeError: no running event loop`.

I am curious as to why this error hasn't been seen until now, as it is fully reproducible on my system.

**Comment by archfan7411** [comment]
I should've noted that this error occurred when testing changes to migrate to discord.py v2, but I'm skeptical that that's the reason it failed.

**Comment by archfan7411** [comment]
Putting this here for posterity: It does appear the issue was correlated to discord.py v2

**Comment by 2br-2b** [comment]
Fixed by eda408a77e6c1899599f2ac3e197d1ec17d0be5e
